clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java

```diff
@@ -202,6 +203,20 @@ public void unsubscription() {
         assertTrue(state.partitionAssignmentNeeded());
     }
 
+    @Test
+    public void pendingUnsubscribeAndOffsetCommit() {
```

Comment:
This is a good start, but I think we need some additional assertions to validate that the behavior of other methods `such as assignment()` takes into account the unsubscribed partitions.